aospy/automate.py

```diff
@@ -257,25 +256,52 @@ def _compute_or_skip_on_error(calc, compute_kwargs):
         return None
 
 
-def _exec_calcs(calcs, parallelize=False, **compute_kwargs):
+def _submit_calcs_on_client(calcs, client, func):
+    """Submit calculations via dask.bag and a distributed client"""
+    logging.info('Connected to client: {}'.format(client))
+    with dask.set_options(get=client.get):
+        return db.from_sequence(calcs).map(func).compute()
+
+
+def _n_workers(calcs):
+    """The number of workers used in a LocalCluster
+
+    An upper bound is set at the cpu_count or the number of calcs submitted,
+    depending on which is smaller.  This is to prevent more workers from
+    being started than needed (but also to prevent too many workers from
+    being started in the case that a large number of calcs are submitted).
+    """
+    return min(cpu_count(), len(calcs))
+
+
+def _exec_calcs(calcs, parallelize=False, client=None, **compute_kwargs):
     """Execute the given calculations.
 
     Parameters
     ----------
     calcs : Sequence of ``aospy.Calc`` objects
     parallelize : bool, default False
         Whether to submit the calculations in parallel or not
+    client : distributed.Client or None
+        The distributed Client used if parallelize is set to True; if None
+        a distributed LocalCluster is used.
     compute_kwargs : dict of keyword arguments passed to ``Calc.compute``
 
     Returns
     -------
     A list of the values returned by each Calc object that was executed.
     """
     if parallelize:
-        pool = multiprocess.Pool()
-        return pool.map(lambda calc:
-                        _compute_or_skip_on_error(calc, compute_kwargs),
-                        calcs)
+        def func(calc):
```

Comment:
Let's add either a docstring or a comment with a sentence that explains why we need this wrapper.